balancer/base/balancer.go

```diff
@@ -101,17 +101,41 @@ func (b *baseBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
 	// addrsSet is the set converted from addrs, it's used for quick lookup of an address.
 	addrsSet := make(map[resolver.Address]struct{})
 	for _, a := range s.ResolverState.Addresses {
-		addrsSet[a] = struct{}{}
-		if _, ok := b.subConns[a]; !ok {
+		// Strip attributes from addresses before using them as map keys. So
+		// that when two addresses only differ in attributes pointers (but with
+		// the same attribute content), they are considered the same address.
+		//
+		// Note that this doesn't handle the case where the attribute content is
+		// different. So if users want to set different attributes to create
+		// duplicate connections to the same backend, it doesn't work. This is
+		// fine for now, because duplicate is done by setting Metadata today.
```

Comment:
Hello @menghanl , we were using attributes to create duplicate connections to our backend because Metadata is deprecated. Is the current advice to use this deprecated API in order to create duplicate backend connections?

Reply:
Yes. Please use the deprecated Metadata field (it should be pretty stable, there's no plan to delete it). Sorry for the trouble.

Reply:
We're looking forward to when the attributes can be used for this again. Thanks for the reply!